### PR TITLE
Fix showing NPE for members of enums with missing identifier

### LINKED ISSUE #28957: Symbol Enter throws NPE for incomplete source
**Description:**
Consider the following source snippets,

```
enum Colour {
    RED,
}
```

```
public enum Colour {
    @
    RED
}
```

For these, the SymbolEnter throws the following exception.

```
java.lang.NullPointerException
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.visit(SymbolEnter.java:1789)
	at org.wso2.ballerinalang.compiler.tree.expressions.BLangConstant.accept(BLangConstant.java:107)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.defineNode(SymbolEnter.java:265)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.defineTypeNodes(SymbolEnter.java:1072)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.defineConstructs(SymbolEnter.java:369)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.visit(SymbolEnter.java:303)
	at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:163)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.defineNode(SymbolEnter.java:265)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.definePackage(SymbolEnter.java:258)
	at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:368)
	at io.ballerina.projects.ModuleCompilationState$3.compile(ModuleCompilationState.java:100)
	at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:323)
	at io.ballerina.projects.PackageCompilation.compileIfRequired(PackageCompilation.java:148)
	at io.ballerina.projects.PackageCompilation.getSemanticModel(PackageCompilation.java:107)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.lambda$semanticModel$0(BallerinaWorkspaceManager.java:196)
	at java.base/java.util.Optional.map(Optional.java:265)
```

### PR DESCRIPTION
**Pull request by dulajdilshan**
## Purpose
$title

Fixes #28957 

## Approach
Skip adding missing enum members to top-level-nodes in compilation-unit
Skip adding missing members to union-type members which are created for enums

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by lochana-chathura — COMMENTED** [review]
Looks good!

Can enable the following tests.

https://github.com/ballerina-platform/ballerina-lang/blob/e344acfaceeddd12db07232619f34397f332e567/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/ParserTestRunner.java#L66-L68

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -3467,6 +3470,9 @@ public BLangNode transform(EnumDeclarationNode enumDeclarationNode) {
         BLangUnionTypeNode bLangUnionTypeNode = (BLangUnionTypeNode) TreeBuilder.createUnionTypeNode();
         bLangUnionTypeNode.pos = bLangTypeDefinition.pos;
         for (Node member : enumDeclarationNode.enumMemberList()) {
+            if (((EnumMemberNode) member).identifier().isMissing()) {
```
Comment (lochana-chathura) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:3473]: `((EnumMemberNode) member).identifier()` can extract to a separate varaible?
Same used in L3476

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 2]
```diff
@@ -3452,6 +3452,9 @@ public BLangNode transform(EnumDeclarationNode enumDeclarationNode) {
             publicQualifier = true;
         }
         for (Node member : enumDeclarationNode.enumMemberList()) {
+            if (((EnumMemberNode) member).identifier().isMissing()) {
```
Comment (lochana-chathura) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:3455]: here also.

**Review by lochana-chathura — APPROVED** [review]
LGTM

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**